apps/api/src/test/java/dk/treecreate/api/exceptionhandling/ExceptionController.java

```diff
@@ -0,0 +1,35 @@
+/*
+    A helper class used by the exception handler testing for the purposes of mocking
+    Throws various exceptions based of the input
+ */
+
+package dk.treecreate.api.exceptionhandling;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+public class ExceptionController
+{
+  @PostMapping("/exception")
+  public void getSpecificException(@RequestParam(name = "exceptionName") String exceptionName,
+                                   @RequestParam(required = false, name = "customMessage") String customMessage)
+  {
+    switch (exceptionName)
```

Comment:
Wouldn't we want to have the ResponseStatusException as being the default case for the switch ?

Reply:
This controller exists only for testing purposes so we can throw exceptions in a controlled way (we don't have to use normal controllers etc).
The only code that utilizes it are test, and the default case would never be triggered (every test wants a specific exception)

Reply:
Oh, I get it now, I thought the controller was some smart new way of handling the exceptions, I was about to ask you about its purpose.